e2e-polybft/e2e/helpers_test.go

```diff
@@ -223,16 +224,61 @@ func checkStateSyncResultLogs(
 	expectedCount int,
 ) {
 	t.Helper()
-	require.Len(t, logs, expectedCount)
+	require.Equal(t, len(logs), expectedCount)
 
 	var stateSyncResultEvent contractsapi.StateSyncResultEvent
 	for _, log := range logs {
 		doesMatch, err := stateSyncResultEvent.ParseLog(log)
-		require.True(t, doesMatch)
 		require.NoError(t, err)
+		require.True(t, doesMatch)
 
 		t.Logf("Block Number=%d, Decoded Log=%+v\n", log.BlockNumber, stateSyncResultEvent)
 
 		require.True(t, stateSyncResultEvent.Status)
 	}
 }
+
+// getCheckpointBlockNumber gets current checkpoint block number from checkpoint manager smart contract
+func getCheckpointBlockNumber(l1Relayer txrelayer.TxRelayer, checkpointManagerAddr ethgo.Address) (uint64, error) {
+	checkpointBlockNumRaw, err := ABICall(l1Relayer, contractsapi.CheckpointManager,
+		checkpointManagerAddr, ethgo.ZeroAddress, "currentCheckpointBlockNumber")
+	if err != nil {
+		return 0, err
+	}
+
+	actualCheckpointBlock, err := types.ParseUint64orHex(&checkpointBlockNumRaw)
+	if err != nil {
+		return 0, err
+	}
+
+	return actualCheckpointBlock, nil
+}
+
+// waitForRootchainEpoch blocks for some predefined timeout to reach target epoch
+func waitForRootchainEpoch(targetEpoch uint64, timeout time.Duration,
+	rootchainTxRelayer txrelayer.TxRelayer, checkpointManager types.Address) error {
+	timer := time.NewTimer(timeout)
```

Comment:
`defer timer.Stop()`
Also, maybe it would be good to create ticker here `ticker := time.NewTicker(time.Second)` and use it in select with just `<- ticker.C:`

Reply:
Fixed https://github.com/0xPolygon/polygon-edge/pull/1360/commits/b511a918a4d25ec3dc7fcfe2d48977bf10d28cf0